google/cloud/bigtable/instance_admin.cc

```diff
@@ -84,63 +84,15 @@ StatusOr<InstanceList> InstanceAdmin::ListInstances() {
   return result;
 }
 
-future<StatusOr<InstanceList>> InstanceAdmin::AsyncListInstances(
-    CompletionQueue& cq) {
-  promise<StatusOr<InstanceList>> instance_list_promise;
-  future<StatusOr<InstanceList>> result = instance_list_promise.get_future();
-  auto client = client_;
-  btadmin::ListInstancesRequest request;
-  request.set_parent(project_name());
-
-  struct Accumulator {
-    std::vector<Instance> instances;
-    std::unordered_set<std::string> failed_locations;
-  };
-
-  return internal::StartAsyncRetryMultiPage(
-             __func__, clone_rpc_retry_policy(), clone_rpc_backoff_policy(),
-             MetadataUpdatePolicy(project_name(), MetadataParamTypes::PARENT),
-             [client](grpc::ClientContext* context,
-                      btadmin::ListInstancesRequest const& request,
-                      grpc::CompletionQueue* cq) {
-               return client->AsyncListInstances(context, request, cq);
-             },
-             std::move(request), Accumulator(),
-             [](Accumulator acc,
-                btadmin::ListInstancesResponse const& response) {
-               std::move(response.failed_locations().begin(),
-                         response.failed_locations().end(),
-                         std::inserter(acc.failed_locations,
-                                       acc.failed_locations.end()));
-               std::move(response.instances().begin(),
-                         response.instances().end(),
-                         std::back_inserter(acc.instances));
-               return acc;
-             },
-             cq)
-      .then([](future<StatusOr<Accumulator>> acc_future)
-                -> StatusOr<InstanceList> {
-        auto acc = acc_future.get();
-        if (!acc) {
-          return acc.status();
-        }
-        InstanceList res;
-        res.instances = std::move(acc->instances);
-        std::move(acc->failed_locations.begin(), acc->failed_locations.end(),
-                  std::back_inserter(res.failed_locations));
-        return res;
-      });
-}
-
 future<StatusOr<btadmin::Instance>> InstanceAdmin::CreateInstance(
     InstanceConfig instance_config) {
   auto cq = background_threads_->cq();
-  return AsyncCreateInstance(cq, std::move(instance_config));
+  return AsyncCreateInstanceImpl(cq, std::move(instance_config));
```

Comment:
We could move `AsyncCreateInstanceImpl` into `CreateInstance` and just have one method.
@coryan Do you think that would be cleaner?

Reply:
Let's just do the remove in this PR and we can cleanup further in future PRs.